File: P4/Napakalaki/src/napakalaki/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;
import java.util.Collections;

/**
 *
 * @author Victor Vallecillo Morilla
 */
public abstract class BadConsequence
{
    String text; // para representar lo que dice un mal rollo
    int levels=Player.MAXLEVEL;    
    
    public BadConsequence(String text, int levels){
        this.text=text;
        this.levels=levels;
    }
   
    public BadConsequence(String text, int levels, int nVisible, int nHidden)
    {  
    }
    
    public BadConsequence(String text, int levels, ArrayList<TreasureKind> tVisible,
            ArrayList<TreasureKind> tHidden)
    {
    }
    
    public BadConsequence(String text, boolean death)
    {
    }
   
    public String getText()
    {
        return text;
    }
    public int getLevels()
    {
        return levels;
    }
    
    
    public void substractVisibleTreasure(Treasure t){}
    
    public void substractHiddenTreasure(Treasure t){}
    
    public boolean isEmpty(){
        return false;
    }
    
    public BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v ,ArrayList<Treasure> h )
    {
        return null;
    }
  
    
    public String toString(){
        return " Text = " + text +
               " Levels = " + Integer.toString(levels);
    }
    
}

File: P4/NapakalakiRuby/pkg/NapakalakiRuby-0.0.1/lib/bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
# encoding: UTF-8
require_relative 'treasure_kind.rb'

module NapakalakiRuby

  class BadConsequence
    #con esto tenemos el get 
  attr_reader :text, :levels, :nVisibleTreasures, :nHiddenTreasures, 
    :specificVisibleTreasures , :specificHiddenTreasures,:death 

    #@@MAXTREASURES=10;

    def initialize(aText, someLevels)
      @text=aText
      @levels=someLevels
    end
=begin
    private_class_method :new    
    #con el self dice que es un metodo de la clase, como es un constructor, solo los constructores son de clase
    def self.newLevelNumberOfTreasures (aText, someLevels, someVisibleTreasures, someHiddenTreasures)
      new(aText, someLevels, someVisibleTreasures, someHiddenTreasures, Array.new, Array.new)
    end  
    def self.newLevelSpecificTreasures (aText, someLevels, someSpecificVisibleTreasures, someSpecificHiddenTreasures)
      new(aText, someLevels, 0, 0, someSpecificVisibleTreasures, someSpecificHiddenTreasures)  
    end
    def self.newDeath (aText)
      new(aText, 0, 0, 0, Array.new, Array.new, true)  
    end
=end
     def isEmpty
     end
   
     def substractVisibleTreasure(t)
     end
   
     def substractHiddenTreasure(t)
     end
  
    def adjustToFitTreasureLists(v, h)
      
    end
    
  
    def to_s
      "Text = #{@text}" + 
      "Levels = #{@levels}"
    end
 
    end
  end
